File: java/openjdk17_spring/src/test/java/benchmark/openjdk17_spring/MongoDbIntegrationTest.java
package benchmark.openjdk17_spring;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.data.mongodb.core.query.Criteria;
import org.springframework.data.mongodb.core.query.Query;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.test.annotation.DirtiesContext;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

@ContextConfiguration(classes = Openjdk17SpringApplication.class)
@DataMongoTest
public class MongoDbIntegrationTest {

    private final String GUID = "1111";
    private final String FIRSTNAME = "TestFirstName";
    private final String LASTNAME = "TestLastName";
    private final String EMAIL = "TestEmail";
    private final String ADDRESS = "TestAddress";

    @Test
    public void saveRetrieveTest(@Autowired CustomerRepository mongoRepository) {

        Customer savedCustomer = new Customer(GUID, FIRSTNAME, LASTNAME, EMAIL, ADDRESS);
        mongoRepository.save(savedCustomer);
        Customer loadedCustomer = mongoRepository.findByGuid(GUID);

        assertNotNull(loadedCustomer);
        assertEquals(GUID, loadedCustomer.guid());
        assertEquals(FIRSTNAME, loadedCustomer.first_name());
        assertEquals(LASTNAME, loadedCustomer.last_name());
        assertEquals(EMAIL, loadedCustomer.email());
        assertEquals(ADDRESS, loadedCustomer.address());

    }
}

File: java/kotlin_spring/src/test/kotlin/benchmark/kotlin_spring/MongoDBIntegrationTest.kt
package benchmark.kotlin_spring

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest
import org.springframework.test.context.ContextConfiguration

@ContextConfiguration(classes = [KotlinSpringApplication::class])
@DataMongoTest
class MongoDBIntegrationTest {

    private val GUID = "1111"
    private val FIRSTNAME = "TestFirstName"
    private val LASTNAME = "TestLastName"
    private val EMAIL = "TestEmail"
    private val ADDRESS = "TestAddress"

    @Test
    fun saveRetrieveTest(@Autowired mongoRepository: CustomerRepository) {
        val savedCustomer = Customer(GUID, FIRSTNAME, LASTNAME, EMAIL, ADDRESS)
        mongoRepository.save(savedCustomer)
        val loadedCustomer = mongoRepository.findByGuid(GUID)
        assertNotNull(loadedCustomer)
        assertEquals(GUID, loadedCustomer!!.guid)
        assertEquals(FIRSTNAME, loadedCustomer!!.first_name)
        assertEquals(LASTNAME, loadedCustomer!!.last_name)
        assertEquals(EMAIL, loadedCustomer!!.email)
        assertEquals(ADDRESS, loadedCustomer!!.address)
    }
}